rewriter/RequiresAncestorKernel.cc

```diff
@@ -0,0 +1,51 @@
+#include "rewriter/RequiresAncestorKernel.h"
+#include "ast/Helpers.h"
+#include "ast/ast.h"
+#include "core/Names.h"
+#include "core/core.h"
+
+using namespace std;
+
+namespace sorbet::rewriter {
+
+namespace {
+
+bool hasAnyRequiresAncestorCall(core::MutableContext ctx, ast::ClassDef *klass) {
+    return absl::c_any_of(klass->rhs, [](const auto &stat) {
+        auto send = ast::cast_tree<ast::Send>(stat);
+        return send && send->fun == core::Names::requiresAncestor() && send->recv.isSelfReference() && send->hasBlock();
```

Comment:
Is it also worth checking that the args are empty? I suppose that'll be caught during type checking so it's maybe not worth catching here.